d2ir/d2ir.go

```diff
@@ -770,7 +770,8 @@ func (m *Map) CreateEdge(eid *EdgeID, refctx *RefContext) (*Edge, error) {
 	if len(common) > 0 {
 		tmp := *refctx.Edge.Src
 		kp := &tmp
-		kp.Path = kp.Path[:len(common)]
+		underscores := countUnderscores(kp.IDA())
```

Comment:
ooo true. Though this should instead be in `trimCommon`. or actually can you combine `resolveUnderscores` and `trimCommon`? This code should be in there.